realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -826,9 +828,16 @@ public void removeChangeListeners() {
      * Notifies all registered listeners.
      */
     void notifyChangeListeners() {
-        realm.checkIfValid();
```

Comment:
`notifyChangeListeners` is always in a safe path (called internally) or preceded by a check  `realm.checkIfValid()`